src/hidra/receiver/datareceiver.py

```diff
@@ -259,6 +260,97 @@ def stop(self):
         self.run_loop = False
 
 
+def run_plugin_thread(
+        plugin_name, plugin_config, target_dir, data_queue, log, event):
+    # Execute all plugin code on a separate thread to protect main thread from
+    # slow or blocking plugins
+    try:
+        plugin_m = import_module("plugins." + plugin_name)
+        plugin = plugin_m.Plugin(plugin_config)
+        plugin.setup()
```

Comment:
if there is more than a single instance of mention, it might help to define a base class; this allows for more accurate exceptions and error handling -> create a base class either for such a module, or the Plugin class

Reply:
Added description